### PR TITLE
added support for docker development

### PR DESCRIPTION
**Pull request by OwenRay**
simply type `docker-compose up` to start developing

### CODE REVIEW

**Review by Evydder — APPROVED** [review]

**Review by camenlx — CHANGES_REQUESTED** [review]
Comments: please introduce goal to use docker, is it to continuous build machine?
introduce some docker and docker-compose install/dependencies to be able to get at "docker-compose up"
at the example small text manual at a file:
https://github.com/zocteam/zeroonecoin/blob/master/doc/docker.md 
and introduce this file at doc bullet at:
https://github.com/zocteam/zeroonecoin/tree/master/doc#building
or
https://github.com/zocteam/zeroonecoin/tree/master/doc#development 

file comments:
.dockerignore
2. release no more exist
6. really need for self ignore?

.gitignore
116. no dash file should exist at all
117. no dash file should exist at all

Dockerfile
use either "\~" or "/root" but same at all files like in docker-compose.yml
16. "RUN mkdir -p /root/.zeroonecore
17. change "\~" into "/root"

I am ok with if auto generated files are to be deleted completely from repo?
autom4te.cache/output.0
autom4te.cache/output.1
autom4te.cache/output.2
autom4te.cache/requests
autom4te.cache/traces.0
autom4te.cache/traces.1
autom4te.cache/traces.2

docker-compose.yml
6. should not be port 10000:10000 ?
9. is source route for repo git clone or mistype of code src?

dockerentry.sh
17. is source route for repo git clone or mistype of code src?
19. is source route for repo git clone or mistype of code src?
22. to rebuild 'make clean' before new 'make'

Missing file with git clone command from https://github.com/zocteam/zeroonecoin, is it Dockerfile or dockerentry.sh or pre-requisite before "docker build" or "docker-compose up" to .:/source ?

**Comment by OwenRay** [comment]
Fixed all but a few; they need some explanation.
> 9. is source route for repo git clone or mistype of code src?

it's the directory where the source code gets mounted inside the docker container.
> 17. is source route for repo git clone or mistype of code src?
> 19. is source route for repo git clone or mistype of code src?

In these cases it does point to the src directory and is meant to do is

> 22. to rebuild 'make clean' before new 'make'

autogen already runs a clean

> Missing file with git clone command from https://github.com/zocteam/zeroonecoin, is it Dockerfile or dockerentry.sh or pre-requisite before "docker build" or "docker-compose up" to .:/source ?

like mentioned before, the source files get mounted inside the container, any changes you make get picked up and trigger a rebuild

**Review by camenlx — APPROVED** [review]
ok nice, going to try this also

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,9 @@
 *.sublime-project
 *.sublime-workspace
+.idea
 todo.txt
 reset-files.bash
 
-!release
```
Comment (camenlx) [inline .gitignore:6]: ok, this is an inline comment, hehehe

**Review by camenlx — APPROVED** [review]

**Pull request merged**